crates/throttle/src/lib.rs

```diff
@@ -3,23 +3,20 @@
 //! to support using a redis-cell equipped redis server to share the throttles
 //! among multiple machines.
 #[cfg(feature = "impl")]
-use mod_redis::{Cmd, FromRedisValue, RedisConnection, RedisError};
+use mod_redis::{RedisConnection, RedisError};
+
 #[cfg(feature = "impl")]
 use once_cell::sync::OnceCell;
-#[cfg(feature = "impl")]
-use redis_cell_impl::{time, MemoryStore, Rate, RateLimiter, RateQuota};
 use serde::{Deserialize, Serialize};
 use std::convert::TryFrom;
-#[cfg(feature = "impl")]
-use std::sync::Mutex;
 use std::time::Duration;
 use thiserror::Error;
 
 #[cfg(feature = "impl")]
 pub mod limit;
-
 #[cfg(feature = "impl")]
-static MEMORY: OnceCell<Mutex<MemoryStore>> = OnceCell::new();
+mod throttle;
```

Comment:
Not sure if the `impl` feature name makes a lot of sense to me, it seems like the main goal is to isolate code that relies on Redis, especially maybe for test purposes? At least this commit seems to help bring the number of guards down a bit.

Reply:
the main goal was to allow the kumo-api-types to crate to import the `ThrottleSpec` type without pulling in redis itself

Reply:
Renamed the `impl` feature to `redis`.

Reply:
I'm not totally sold on renaming this, because now the feature name redis implies that it enables only redis related functions, but the current state of this PR also ties the in-memory implementation to the redis feature.

This is somewhat of an abstract point, because it doesn't really change anything today.

It would be nice though for the traffic-gen crate to be able to pull in the in-memory throttle impl without also pulling in the redis deps.